### PR TITLE
Improve depends debuggability

### PR DESCRIPTION
**Pull request by dongcarl**
Adds make targets for each package for each stage, e.g.

```sh
make zeromq_configured
```

### CODE REVIEW

**Comment by MarcoFalke** [comment]
ACK 6d44c5ebf97af4b357079fe4bc2130f98e1d0fd2 (Haven't looked at the code changes, but adding this feature makes sense)

**Comment by fanquake** [comment]
Concept ACK

**Comment by laanwj** [comment]
@theuni can you take a look please

File: depends/funcs.mk [hunk 1]
```diff
@@ -213,6 +213,14 @@ $(1): | $($(1)_cached_checksum)
 
 endef
 
+stages = fetched extracted preprocessed configured built staged postprocessed cached cached_checksum
+
+define ext_add_stages
+$(foreach stage,$(stages),
+          $(1)_$(stage): $($(1)_$(stage))
```
Comment (ryanofsky) [inline depends/funcs.mk]: This code is pretty tucked away and it seems like it nobody is going to know these debug targets exist if they aren't documented somewhere. Ideally I think you'd be able to go to the depends directory and run "make help" and get basic usage information that mentions these. Maybe something to implement in a future PR.

For now at least the new targets can be listed with some stackoverflow shell magic:

```
$ make -pRrq : 2>/dev/null | awk -v RS= -F: '/^# File/,/^# Finished Make data base/ {if ($1 !~ "^[#.]") {print $1}}' | egrep -v '^[^[:alnum:]]' | sort
all
bdb
bdb_built
bdb_cached
bdb_cached_checksum
bdb_configured
bdb_extracted
bdb_fetched
bdb_postprocessed
bdb_preprocessed
bdb_staged
boost
boost_built
boost_cached
boost_cached_checksum
boost_configured
boost_extracted
boost_fetched
boost_postprocessed
boost_preprocessed
boost_staged
cached
check-packages
check-sources
clean
clean-all
dbus
dbus_built
dbus_cached
dbus_cached_checksum
dbus_configured
dbus_extracted
dbus_fetched
dbus_postprocessed
dbus_preprocessed
dbus_staged
download
download-linux
download-one
download-osx
download-win
expat
expat_built
expat_cached
expat_cached_checksum
expat_configured
expat_extracted
expat_fetched
expat_postprocessed
expat_preprocessed
expat_staged
fontconfig
fontconfig_built
fontconfig_cached
fontconfig_cached_checksum
fontconfig_configured
fontconfig_extracted
fontconfig_fetched
fontconfig_postprocessed
fontconfig_preprocessed
fontconfig_staged
freetype
freetype_built
freetype_cached
freetype_cached_checksum
freetype_configured
freetype_extracted
freetype_fetched
freetype_postprocessed
freetype_preprocessed
freetype_staged
install
libX11
libX11_built
libX11_cached
libX11_cached_checksum
libX11_configured
libX11_extracted
libX11_fetched
libX11_postprocessed
libX11_preprocessed
libX11_staged
libXau
libXau_built
libXau_cached
libXau_cached_checksum
libXau_configured
libXau_extracted
libXau_fetched
libXau_postprocessed
libXau_preprocessed
libXau_staged
libXext
libXext_built
libXext_cached
libXext_cached_checksum
libXext_configured
libXext_extracted
libXext_fetched
libXext_postprocessed
libXext_preprocessed
libXext_staged
libevent
libevent_built
libevent_cached
libevent_cached_checksum
libevent_configured
libevent_extracted
libevent_fetched
libevent_postprocessed
libevent_preprocessed
libevent_staged
libxcb
libxcb_built
libxcb_cached
libxcb_cached_checksum
libxcb_configured
libxcb_extracted
libxcb_fetched
libxcb_postprocessed
libxcb_preprocessed
libxcb_staged
miniupnpc
miniupnpc_built
miniupnpc_cached
miniupnpc_cached_checksum
miniupnpc_configured
miniupnpc_extracted
miniupnpc_fetched
miniupnpc_postprocessed
miniupnpc_preprocessed
miniupnpc_staged
native_protobuf
native_protobuf_built
native_protobuf_cached
native_protobuf_cached_checksum
native_protobuf_configured
native_protobuf_extracted
native_protobuf_fetched
native_protobuf_postprocessed
native_protobuf_preprocessed
native_protobuf_staged
openssl
openssl_built
openssl_cached
openssl_cached_checksum
openssl_configured
openssl_extracted
openssl_fetched
openssl_postprocessed
openssl_preprocessed
openssl_staged
protobuf
protobuf_built
protobuf_cached
protobuf_cached_checksum
protobuf_configured
protobuf_extracted
protobuf_fetched
protobuf_postprocessed
protobuf_preprocessed
protobuf_staged
qrencode
qrencode_built
qrencode_cached
qrencode_cached_checksum
qrencode_configured
qrencode_extracted
qrencode_fetched
qrencode_postprocessed
qrencode_preprocessed
qrencode_staged
qt
qt_built
qt_cached
qt_cached_checksum
qt_configured
qt_extracted
qt_fetched
qt_postprocessed
qt_preprocessed
qt_staged
xcb_proto
xcb_proto_built
xcb_proto_cached
xcb_proto_cached_checksum
xcb_proto_configured
xcb_proto_extracted
xcb_proto_fetched
xcb_proto_postprocessed
xcb_proto_preprocessed
xcb_proto_staged
xextproto
xextproto_built
xextproto_cached
xextproto_cached_checksum
xextproto_configured
xextproto_extracted
xextproto_fetched
xextproto_postprocessed
xextproto_preprocessed
xextproto_staged
xproto
xproto_built
xproto_cached
xproto_cached_checksum
xproto_configured
xproto_extracted
xproto_fetched
xproto_postprocessed
xproto_preprocessed
xproto_staged
xtrans
xtrans_built
xtrans_cached
xtrans_cached_checksum
xtrans_configured
xtrans_extracted
xtrans_fetched
xtrans_postprocessed
xtrans_preprocessed
xtrans_staged
zeromq
zeromq_built
zeromq_cached
zeromq_cached_checksum
zeromq_configured
zeromq_extracted
zeromq_fetched
zeromq_postprocessed
zeromq_preprocessed
zeromq_staged
zlib
zlib_built
zlib_cached
zlib_cached_checksum
zlib_configured
zlib_extracted
zlib_fetched
zlib_postprocessed
zlib_preprocessed
zlib_staged
```

https://stackoverflow.com/questions/4219255/how-do-you-get-the-list-of-targets-in-a-makefile/45843594#45843594

**Review by ryanofsky — APPROVED** [review]
ACK 6d44c5ebf97af4b357079fe4bc2130f98e1d0fd2

**Pull request merged**